test/unit/microcosm/patch.test.js

```diff
@@ -98,6 +98,72 @@ describe('Microcosm::patch', function () {
       // the new state produced by a child.
       expect(child.state.count).toEqual(4)
     })
+
+    it('does not strip away parent inherited state', function () {
+      const parent = new Microcosm()
+      const child = parent.fork()
+
+      parent.addDomain('top', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      child.addDomain('bottom', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      child.patch({ bottom: true })
+
+      expect(child).toHaveState('top', false)
+      expect(child).toHaveState('bottom', true)
+    })
+
+    it('does not strip away parent inherited state when the parent patches', function () {
+      const parent = new Microcosm()
+      const child = parent.fork()
+
+      parent.addDomain('top', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      child.addDomain('bottom', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      parent.patch({ top: true })
+
+      expect(parent).toHaveState('top', true)
+      expect(child).toHaveState('top', true)
+    })
+
+    it('patching on a parent does not reset state of children', function () {
+      const parent = new Microcosm()
+      const child = parent.fork()
+
+      parent.addDomain('top', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      child.addDomain('bottom', {
+        getInitialState() {
+          return false
+        }
+      })
+
+      parent.patch({ bottom: true })
+
+      expect(parent).not.toHaveState('bottom')
+      expect(child).toHaveState('bottom', false)
+    })
```

Comment:
It's an internal Microcosm thing. `addons/jest-matchers`. They make my life easier, and I want to share them with others. I just don't think they're quite ready yet.